signer/src/main.rs

```diff
@@ -12,10 +20,61 @@ fn get_connection_pool() -> sqlx::PgPool {
         .unwrap()
 }
 
+/// Command line arguments for the signer.
+#[derive(Debug, Parser)]
+#[clap(name = "sBTC Signer")]
+struct SignerArgs {
+    /// Optional path to the configuration file. If not provided, it is expected
+    /// that all parameters are provided via environment variables.
+    #[clap(short = 'c', long, required = false)]
+    config: Option<PathBuf>,
+}
+
 #[tokio::main]
-async fn main() {
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Initialize logging
     sbtc::logging::setup_logging("info,signer=debug", false);
 
+    // Parse the command line arguments.
+    let args = SignerArgs::parse();
+
+    // Initialize the signer context.
+    let context = SignerContext::init(args.config)?;
+
+    // Run the Stacks event observer and Ctrl-C watcher concurrently.
+    let _ = tokio::join!(
+        run_stacks_event_observer(&context),
+        run_ctrl_c_watcher(&context),
+        run_libp2p_swarm(&context),
+    );
+
+    Ok(())
+}
+
+/// Runs the Ctrl-C watcher.
+async fn run_ctrl_c_watcher(ctx: &impl Context) -> Result<(), Error> {
+    tokio::signal::ctrl_c().await?;
+    ctx.signal_shutdown()?;
+    Ok(())
+}
```

Comment:
Why not just take ownership of a signal sender here?

Reply:
> Why not just take ownership of a signal sender here?

Do you mean a `Sender<SignerSignal>`? If so, it was just slightly more ergonomic to provide a convenience method. But I also felt that it was a bit superfluous, so I'll remove `signal_shutdown()` and just use `signal_send()`. `tokio::sync::broadcast::Sender<T>` can be used concurrently via an `Arc` (added the `Arc` now), so doing that via the `SignerContext` is just fine, but receivers need to be created for each thread.

>Also, we might want to pre-emtively rename this function since we'll probably want to watch for other signals down the line (like SIGTERM and SIGHUP).

Done, also added support for `SIGTERM`, `SIGHUP` and `SIGINT` (and kept ctrl-c for non-unix).